### DISCUSSION TITLE
Run optimize through a UI

### DISCUSSION THREAD

**Discussion by MosaSalim99** [Tunny-gh/Tunny_in_Open_Sea#199] [General]
Hi! First of all great Plugin! I am creating a tool that calculates utilization, carbon emission and cost of roof trusses and then outputs the best solution based on some inputs. I created a UI through human UI to get the inputs from the user but i want to include a few inputs to go to the tunny component UI. I used MetaHopper to refrence the tunny component and then in to python, but i cant figureout how to input the values i want, which are:
(Optimize tab)
Number of trials
Study Name
RunOptimize
(Visualize tab)
Open Optuna-Dashboard
(Output tab)
Target Study
All trials
![tunny](https://github.com/hrntsm/Tunny/assets/143790285/3681af6b-0cb5-4b06-9783-8885374bb925)

**Comment by hrntsm** [comment]
Thanks @MosaSalim99  for your interest.
Sorry, I don't know much about Metahopper so I don't yet understand what kind of question you are asking.
Can you give me more details?

**Reply by MosaSalim99** [reply]
Shure!
The MetaHopper isnt that important i just use it so i can connect my Tunny component with Python. 
I want to access the Tunny UI (when i doubbleclick Tunny) and input the values listed earlier, however i want to do it without opening the Tunny UI.

**Reply by hrntsm** [reply]
Tunny is optimizing by creating an Optuna class and executing the RunSolver method, but Tunny's optimization may not work well because it is based on the TunnyUI assumption.
https://github.com/hrntsm/Tunny/blob/d101e78217946552313cce73b3440d1475873023/Tunny/Solver/Optuna.cs#L22-L62
